pinc/Project.inc

```diff
@@ -18,24 +18,74 @@ class UTF8ConversionException extends Exception
 
 class ProjectException extends Exception
 {
+    // all project errors extend from here and in range 100-199
+    public function __construct($message, $code = 100)
```

Comment:
For this and similar, messages to be added later?

Reply:
The message is generally put in when the exception is thrown because it might include data from the context but it wouldn't hurt to put in default messages for all cases.

Reply:
Thanks for the explanations.